service/bblfsh/bblfsh.go

```diff
@@ -42,22 +44,95 @@ func (s *Service) GetChanges(ctx context.Context, req *lookout.ChangesRequest) (
 	}
 
 	return &ChangeScanner{
-		underlying:    changes,
-		ctx:           ctx,
-		client:        s.client,
-		purgeContents: !wantContents,
+		underlying: changes,
+		BaseScanner: BaseScanner{
+			ctx:           ctx,
+			client:        s.client,
+			purgeContents: !wantContents,
+		},
 	}, nil
 }
 
-type ChangeScanner struct {
-	underlying    lookout.ChangeScanner
+func (s *Service) GetFiles(ctx context.Context, req *lookout.FilesRequest) (
+	lookout.FileScanner, error) {
+	wantContents := req.WantContents
+	if req.WantUAST {
+		req.WantContents = true
+	}
+
+	files, err := s.files.GetFiles(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if !req.WantUAST {
+		return files, nil
+	}
+
+	return &FileScanner{
+		underlying: files,
+		BaseScanner: BaseScanner{
+			ctx:           ctx,
+			client:        s.client,
+			purgeContents: !wantContents,
```

Comment:
Could you help me understand if `purgeContents` can even happen here? `wantContents` seems to be always True, if an execution has reached here after L60.

Reply:
line 58: `wantContents := req.WantContents` creates new boolean with original client request.
line 60: don't change it.